google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1077,7 +1077,7 @@ private void replenishPool() {
         // If we have gone below min pool size, create that many sessions.
         int sessionCount = options.getMinSessions() - (totalSessions() + numSessionsBeingCreated);
         if (sessionCount > 0) {
-          createSessions(getAllowedCreateSessions(sessionCount));
+          createSessions(getAllowedCreateSessions(sessionCount), false);
```

Comment:
So does this mean only when we initialize the pool, we will distribute requests across different channels? For rest, we don't.

Reply:
Yes, that is correct. The `SessionClient` will automatically make sure that consecutive requests will be executed on different channels round robin, so that the distribution over channels will still be maintained if the session pool continues to request additional sessions.

This means that the pool initialization will distribute the `MinSessions` sessions over all available channels. Each time the pool increases the number of sessions, that batch of new sessions will all be bound to one channel, but each new batch will be bound to a different channel. So in the end the sessions will still be distributed over all available channels.